Added language in codeblocks

ktsrivastava29 opened the pull request:
I'm adding languages to codeblocks in md files. There are a few code blocks that don't have a language for their lexer, and Pygments isn't always smart enough to figure out what language is intended.
The language should be stated as follows: 
````
```language

code

```
````
fix: https://github.com/plone/volto/issues/3090
also see: https://github.com/plone/volto/pull/3155

stevepiercy commented on docs/source/recipes/creating-project.md:
```diff
@@ -11,30 +11,34 @@ html_meta:
 For using Volto for a project (i.e. use Volto as a library), You should use Volto's project generator `@plone/generator-volto`. It's a boilerplate generator based in Yeoman that will provide you with the basic files and folder structure to bootstrap a Volto site. In addition to bootstrapping standalone Volto projects, it can also bootstrap Volto addons.
 
 1. Open a terminal and execute:
```
For this numbered list, the syntax is a little tricky. Any content in a list must be indented. I also prefer to indent everything 4 spaces and have it aligned, including the content after a number or bullet. For example:

````
1.  Open a terminal and execute:

    ```shell
    npm install -g yo @plone/generator-volto
    yo @plone/volto
    ```
````

...will render nicely with the code-block within the context of item 1. Please apply the same to this list, and use `make docs-livehtml` to watch the changes as you edit them in real time.

stevepiercy replied:
The code-block still does not indent properly. See the build preview:

https://deploy-preview-3165--volto.netlify.app/recipes/creating-project.html

Compare that code block to the code-blocks in https://deploy-preview-3165--volto.netlify.app/getting-started/install.html#install-nvm-nodejs-version-manager.

You can also look at the source of that file if you need further guidance.

stevepiercy commented on docs/source/recipes/creating-project.md:
````diff
@@ -11,30 +11,34 @@ html_meta:
 4. The project is ready to be started, `@plone/generator-volto` already has run the
    dependencies installations for you.
+
 ```
````
````suggestion
```shell
````

stevepiercy commented on docs/source/recipes/creating-project.md:
````diff
@@ -11,30 +11,34 @@ html_meta:
    dependencies installations for you.
+
 ```
 $ yarn start
````
Purge all prompts.

```suggestion
yarn start
```

stevepiercy commented on docs/source/recipes/i18n.md:
```diff
@@ -104,20 +104,25 @@ This approach can be best explained with an example: Assume you have a component
 To translate the `alt` attribute, you have to do the following steps:
 
 1. Import the following necessary methods:
```
See above comment about indentation of list items.

stevepiercy commented on docs/source/recipes/i18n.md:
````diff
@@ -127,14 +132,17 @@ class TeaserImage extends Component {
+
 4. As last step, the method can be used like this:
+
 ```
````
````suggestion
```html
````

stevepiercy commented on docs/source/recipes/i18n.md:
````diff
@@ -127,14 +132,17 @@ class TeaserImage extends Component {
 ```
+
 Since you now have another prop available, it has to be to properly defined in the propTypes:
 ```
````
I'm not sure, but I think this is `jsx`.

````suggestion
```jsx
````

stevepiercy commented on docs/source/recipes/i18n.md:
````diff
@@ -152,7 +160,7 @@ $ yarn i18n
 This will generate the following output:
 
-```
+```shell
````
This is output, not a shell command. For output, we use `console`.

````suggestion
```console
````

stevepiercy commented on docs/source/recipes/i18n.md:
```diff
@@ -212,23 +220,31 @@ When you done your translation resource at Transifex, you need to add the locale
 To get translate the latest translation changes, you have to do the following steps:
 
 1. You can pull the latest changes for a specific language, for example `de` from Transifex for *Volto project*, by this command:
+
```
See above comment about indenting list items with code-blocks.

stevepiercy commented on docs/source/recipes/i18n.md:
````diff
@@ -238,24 +254,34 @@ This command generate again the ``locales/de.json`` file updated, it is needed f
+
 Also you need define the new locale into *'locales const'* as the follow:
+
 ```
````
This one needs a language. I am not sure which. Is it Python?

stevepiercy commented on docs/source/recipes/i18n.md:
````diff
@@ -238,24 +254,34 @@ This command generate again the ``locales/de.json`` file updated, it is needed f
+
 4. You new to modify the ``src/constants/Languages.js`` file and add a new line at *'module.exports'* section as the follow:
+
 ```
````
````suggestion
```js
````

stevepiercy commented on docs/source/recipes/lazyload.md:
````diff
@@ -35,7 +35,7 @@ You can find the complete `@loadable/component` documentation here: https://load
 You can check the code splitting state by using the included bundle analyzer:
 
 ```bash
````
We want to replace all instances of the bash language with shell so that we are shell-neutral.

````suggestion
```shell
````

stevepiercy reviewed (changes requested):
Please see individual comments.

This will also need a review from a maintainer.

stevepiercy commented:
You will also need to resolve merge conflicts with `master` again.

ktsrivastava29 commented:
@stevepiercy I've now fixed the indent and have also resolved the conflicts

stevepiercy commented:
@ktsrivastava29 please check the preview build and note the list is now broken.

https://deploy-preview-3165--volto.netlify.app/recipes/creating-project.html

Please build the docs locally, and inspect a preview locally, before pushing more commits.

ktsrivastava29 commented:
@stevepiercy I've fixed that, please let me know how can I make it better

stevepiercy commented:
@ktsrivastava29 again, please check the preview. And not just for this one file, but for all files you have touched.

https://deploy-preview-3165--volto.netlify.app/recipes/creating-project.html

Before pushing any more commits to this pull request, and asking for another review, first build the docs locally and preview them to ensure that they render correctly.

ktsrivastava29 commented:
@stevepiercy I sincerely apologize for bothering you again; actually, I went through each file I edited and fixed the indent where necessary; locally, everything was great, but when I pushed, the test failed; could you kindly assist me in resolving this error?

stevepiercy commented:
@ktsrivastava29 the failure can be resolved by rebasing your branch onto `master`, then force pushing it. We recently added the linkcheck requirement in GitHub Actions, but it is not in your branch's `Makefile`.

stevepiercy reviewed (changes requested):
When you build the docs locally, please check warnings and visually inspect the build before pushing commits and asking for my review. I have to assume you are not doing that, based on what I see in this review round. This PR still needs work. It is not OK to repeatedly say that you have addressed suggestions, but have actually not addressed them.

nileshgulia1 reviewed (commented):
@ktsrivastava29 There's a simple command `make docs-livehtml` https://github.com/plone/volto/blob/master/Makefile#L127 which will save you a lot of time while developing on this PR. Please use it before pushing further commits.

ktsrivastava29 commented:
@stevepiercy I've fixed all and double-checked everything; I apologise for taking up so much of the time. Actually previously, I was receiving some HTTPS errors, which took me a day to resolve, and then I got an error due to the improper directory name; after resolving all of these, I was able to go through the suggested changes and then modify it.

stevepiercy commented on docs/source/recipes/creating-project.md:
````diff
@@ -10,31 +10,37 @@ html_meta:
+    cd myvoltoapp
+    ```
+
+4.  The project is ready to be started, `@plone/generator-volto` already has run the
````
Let's keep the sentence together on one line.

```suggestion
4.  The project is ready to be started, `@plone/generator-volto` already has run the dependencies installations for you.
```

stevepiercy commented on docs/source/recipes/creating-project.md:
````diff
@@ -10,31 +10,37 @@ html_meta:
+    ```
+
+4.  The project is ready to be started, `@plone/generator-volto` already has run the
    dependencies installations for you.
````
```suggestion
```

stevepiercy commented on docs/source/recipes/environment-variables.md:
````diff
@@ -24,21 +24,21 @@ Let's assume you have your site in http://localhost:55001/Plone
 In order to simulate it, you could launch the backend like:
 
-```
+```shell
````
Please also wrap lines 12 and 17 above as shell commands.

stevepiercy commented on lib64:
```diff
@@ -0,0 +1 @@
+lib
```
This file should not be committed to the repository. Please remove.

stevepiercy replied:
This file still exists in your PR. Please remove it from the PR.

stevepiercy reviewed (changes requested):
We're almost there! All the lists look good now. There is a file that should not be included in this PR, and a couple of other minor changes, a few which were overlooked.

ktsrivastava29 commented:
@stevepiercy I've made the changes you recommended; please take a look now.

stevepiercy commented on .gitignore:
```diff
@@ -9,6 +9,7 @@ junit.xml
 eslint.xml
 yarn-error.log
 build
+lib64
```
This is Python related, not Node. Please move this line under the #Python section, following its sorting and leading slash patterns.

stevepiercy reviewed (changes requested):
One more change, please.

ktsrivastava29 commented:
@stevepiercy Oh, I'm sorry, I wrongly placed it; I've fixed it now.

ktsrivastava29 commented:
@stevepiercy Previously, only one test failed; now, two have failed. I haven't made any changes since the last push, except for the minor change in the gitignore file that you suggested.

stevepiercy commented:
The [failing docs build](https://github.com/plone/volto/runs/5708574282?check_suite_focus=true) will be fixed in https://github.com/plone/volto/pull/3196. We can ignore that failure.

For the [failing core sandbox tests build](https://github.com/plone/volto/runs/5708574126?check_suite_focus=true#step:8:738), I am not sure. @tiberiuichim or @sneridagh, have you got any idea? I restarted that failure to see if it was just a temporary connection issue. Let's see if it fails again. I restarted that failed build for my PR, and it just now passed.

stevepiercy commented:
We just merged https://github.com/plone/volto/pull/3196, so the docs build should now pass, I think.

stevepiercy commented:
meh, nm, docs build still breaks because this PR has not switched to use plain old `docs-linkcheck`. The docs build failure may be ignored.

nileshgulia1 commented:
@ktsrivastava29 You can try rebasing your PR with master.

stevepiercy reviewed (approved):
Excellent. Thank you!

Would you please also either merge `master` into your branch or rebase your branch on `master` and force push? Then that should get the failing docs build to pass, and should allow you to merge your own pull request.

stevepiercy commented:
Oops, actually, it looks like the docs test now passes. Nevermind about rebasing or merging. Let's wait for all green, then you can merge.

ktsrivastava29 commented:
@stevepiercy @nileshgulia1 Thank you; this is my third PR in this organisation, and I'm overjoyed as after 10 days it finally got approved. I'm learning a lot from your experiences. I'm eager to learn and contribute as much as I can.

stevepiercy commented:
@ktsrivastava29 please do the honor of merging your PR.

ktsrivastava29 commented:
@stevepiercy Sorry, you will have to do it because the base branch restricts merging to me, perhaps only the core team is authorized to merge into master.
